File: fizzbuzz/Elixir/fizzbuzz.ex
# Use elixir rnd.ex
# Classic fizzbuzz problem from 1 to 100

fizzbuzz = fn
    (0, 0, _) -> "FizzBuzz\n"
    (0, _, _) -> "Fizz\n"
    (_, 0, _) -> "Buzz\n"
    (_, _, n) -> Integer.to_string(n) <> "\n"
end

remain = fn (n) ->
    fizzbuzz.(rem(n, 3), rem(n, 5), n)
end

IO.puts Enum.map(1..100, remain)

File: fizzbuzz/Haskell/fizzbuzz.hs
-- Compile using ghc --make rand.hs
-- Classic fizzbuzz problem from 1 to 100

fizzbuzz n = case (rem n 3, rem n 5) of
    (0, 0) -> "FizzBuzz"
    (0, _) -> "Fizz"
    (_, 0) -> "Buzz"
    (_, _) -> show n

main = mapM_ (putStrLn . fizzbuzz) [1..100]
